### PR TITLE
Add react-jsonschema-form with material design components

### LINKED ISSUE #7: Research using JSON Schema for tools definitions
The current version of the tools definition uses a custom implementation based on [Checkit](https://github.com/tgriesser/checkit) and other custom tools. This has multiple disadvantages like not being able to use standard tooling and having to implement any limitation the library has (at the moment we have code that extends it).

Form definitions as are currently called have everything a tool needs to have, including the text we display for the tools description and how we store the data, here's an example https://github.com/debtcollective/dispute-tools/blob/master/lib/data/form-definitions/credit-report-dispute.js.

To improve what we have and reduce the amount of code we maintain, we want to use [JSON Schema](https://json-schema.org/) to define our tools and use it as the source of truth for the tool.

As part of the research, we expect to define:

- How we will store the data in the database
- How we will handle validations
- How we will render forms and wizards from the tool definition

## Acceptance Criteria

- Create a spec that includes a way for us to adopt JSON schema as our tool definition framework.

### LINKED ISSUE #8: Port General Debt Dispute tool to JSON Schema
We need to port the [General Debt Dispute tool definition](https://github.com/debtcollective/dispute-tools/blob/master/lib/data/form-definitions/general-debt-dispute.js) to JSON Schema.

## Acceptance Criteria

- General Debt Dispute tool should work using JSON Schema

### PR DESCRIPTION
**Pull request by duranmla**
**What:** Integrate [react-jsonschema-form](https://github.com/mozilla-services/react-jsonschema-form) with customizations using Material Design while porting [Port General Debt Dispute tool to JSON Schema](https://github.com/debtcollective/disputes/issues/8)

**Why:**
Closes #7 
Closes #8 

**How:**
- Add [react-jsonschema-form](https://github.com/mozilla-services/react-jsonschema-form)
- Add [Material UI](https://material-ui.com/)
- Port **"General Debt Dispute"** to JSON Schema (re #8)
- Port **"General Debt Dispute"** to UI schema defined by the react-jsonschema-form (re #8)
- Map all inputs within General Debt Dispute to their pairs on Material Design (We have field text with different types, selects, and date picker)
- Add custom dependency to solve mask input with currency [_(as suggested here)_](https://material-ui.com/demos/text-fields/#complementary-projects)
- Add custom dependency to solve date pickers as material _[(as suggested here)](https://material-ui.com/demos/pickers/#complementary-projects)_

**Screenshots:**
image

**Note:**
We do not introduce new tests to perform submits as the code basically works by using correctly the tool we choose, nevertheless, it worth to add E2E in order to make sure the definition allows us to submit the values after being filled properly. Which will be related to #6 and having some backend to send the data to

As we still not have in place the linting this code may need formatted after https://github.com/debtcollective/dispute-tools/issues/175

### CODE REVIEW

**Comment by orlando** [comment]
@duranmla there's a 404 when going to the disputes link, I would like to have it working to try it live.

Let's pair for a bit tomorrow and review this PR

**Comment by duranmla** [comment]
Let's talk about this today at 20:00 (GMT+2) because I am a bit lost regarding how and where to test this in order to see it failing. https://github.com/debtcollective/disputes/pull/56/commits/46a4716792e7728ed5298949157dd0ab8fe53cfb you can see when I added the file `packages/front-end/pages/disputes/general.js` that suppose to add the page as the documentation says.

Nevertheless, I realize that we have two frontend packages one under `app/` and one under `packages/` so maybe I use the wrong one, but I do not know where are you testing it.

**Comment by duranmla** [comment]
Working on porting the next form schema I realize the issue you are facing, seems like the whole code I should place it within `apps/frontend/` folder instead `packages/front-end` (most likely) so introducing that update should make it work as expected.

I will do it today at the time I have said before. 👌

**Comment by duranmla** [comment]
I have updated, nevertheless, I realize with my latest change I break a test (due to date pickers) I need to update before merging this PR. But now you can check the code working by running locally

File: apps/frontend/components/FieldTemplate.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import get from "lodash/get";
```
Comment (orlando) [inline apps/frontend/components/FieldTemplate.js]: @duranmla I think we can use just one lodash import by using https://github.com/lodash/babel-plugin-lodash and https://github.com/lodash/lodash-webpack-plugin. We can optimize this later.

File: apps/frontend/components/PlainTemplate.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    label,
+    help,
+    required,
+    description,
+    errors,
+    children
+  } = props;
+
+  return (
+    <div data-testid="plain-template" className={classNames}>
```
Comment (orlando) [inline apps/frontend/components/PlainTemplate.js]: @duranmla let's keep https://www.npmjs.com/package/babel-plugin-jsx-remove-data-test-id in mind when we start deploying to production

**Pull request merged**